src/RenderMethods/RenderEditorFieldTemplate.tsx

```diff
@@ -0,0 +1,68 @@
+import type { FieldTemplateProps } from '@rjsf/core';
+import { useVerna } from '../context/VernaContextProvider';
+import { makeid } from '../utils';
+import React from 'react';
+import {
+  addItemToSchema,
+  addItemToUiSchema,
+  addSection,
+  removeItem,
+  removeSection,
+} from '../context/InteractionMethods';
+import ShowCaseWidgetProps from '../types/Widgets';
+import DropZone from '../Widgets/DropZone';
+
+// This component is used to render every field
+// Its purpose here is to add functionality common to every of those such as add or remove
+// It will be used to add drag and drop area later
+
+export default function RenderEditorFieldTemplate({ id, schema, children }: FieldTemplateProps) {
+  const verna = useVerna();
+  const isRoot = id === 'root';
+  const isSection = id.split('_').length === 2 && !verna.selector;
+  const ownProperties = Object.keys(schema.properties || {}).length > 0;
+  const canAddField = schema.type !== 'object';
+  const canAddSection = isSection && !isRoot && !verna.selector;
+
+  if (!verna.isEditor) return children;
```

Comment:
Move this after `addItem` function in order to group the code related to the render at the same place.